src/core/Akka.Cluster/Cluster.cs

```diff
@@ -140,7 +140,11 @@ public Cluster(ActorSystemImpl system)
             _readView = new ClusterReadView(this);
 
             // force the underlying system to start
-            _clusterCore = GetClusterCoreRef().Result;
+            // and hard block the current thread
+            var clusterCoreTask = Task.Run(GetClusterCoreRef);
```

Comment:
Are there any flags we should consider adding here? i.e. would it be better to do `Task.Factory.StartNew` with `LongRunning` and `DenyChildAttach`?

Reply:
`_clusterCore = GetClusterCoreRef().Result` was always locking and I think its still is.

I think what happend is that after the Ask got `improved` the current thread gets used for the actorcell dispatcher.
I don't know how/where exactly, but the child-actors of cluster-daemon are using it.

´Task.Run(GetClusterCoreRef)´ should force the ask on a different thread, 
this is the only thing that matters, that it the current thread does not leak.

Reply:
> I think what happend is that after the Ask got improved the current thread gets used for the actorcell dispatcher.

Interesting. Would there be other implications if this is true?

cc: @Aaronontheweb

Reply:
The actor system extensions implementation and the locking cluster constructor are both just anti-patterns 
and we need to rework it in the future.
https://github.com/akkadotnet/akka.net/discussions/5447

Reply:
Maybe its just the "ConfigureAwait(false)" of 
https://github.com/akkadotnet/akka.net/blob/44c51f6786df0f97914381bd1c8b7f84013f7c0e/src/core/Akka.Cluster/Cluster.cs#L155

The Cluster Extension is called by the ClusterActorRefProvider and it is created in the ActorSystem Startup.
The thread is the same as of the ActorSystem creator and with the normal ForkJoinExecutor they will never mix.
But with the ChannelExecutor that uses the normal ThreadPool of dotnet, the awaiting thread can be used for an ActorCell. 

``` 
var task = Task.Run(...);
task.Wait();
``` 
I hope that simply resolve the problem.

`task.Result` blocked at .net 4.5 for sure, 
I don't know why/how the thread gets now reused.